app/controllers/subscriptions_controller.rb

```diff
@@ -15,6 +15,7 @@ def new
 
     if params[:email]
       @email = params[:email]
+      @subscription = Subscription.find_by(email: @email)
```

Comment:
Yes, good thinking, you could use that token.

However it's worth asking ourselves if we really need this element. As you just said, "People will most likely be coming from an alert email if they see that", so why show it again?